packages/core/test/signal.test.tsx

```diff
@@ -160,6 +160,213 @@ describe("effect()", () => {
 
 		expect(fn).to.throw(/Cycle detected/);
 	});
+
+	it("should allow disposing the effect multiple times", () => {
```

Comment:
sorry for a random comment from a random person, was just reading through this interesting code

but this test appears to be showing that disposing the effect multiple times is not allowed

Reply:
And thanks a bunch for looking through the code! The [current main branch version](https://github.com/preactjs/signals/blob/main/packages/core/src/index.ts) has been iterated on further quite a bit, so I recommend checking it out. In both cases apologies for the code's readability, there are some comments sprinkled there but they probably convey all the tricks there 😅

Reply:
Argh, update: I meant to say "they probaby **don't** convey all the tricks there".